feat : create questionnaire form components

keithCuniah opened the pull request:
# Description
form and components for the questionnaire in feedback task

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

See #2672 
**Type of change**
- add the form and subcomponent (rating, textArea and single label)
- manage error when question with a required status does not have any answer

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Feedback task (new task independent of previous task)

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

keithCuniah committed:
layout for llm dataset

keithCuniah committed:
Merge branch 'develop' into feat/create-layout-instructions-task

keithCuniah committed:
Merge branch 'develop' into feat/create-layout-instructions-task

keithCuniah committed:
remove obsolete file

keithCuniah committed:
custom directive to show required field

keithCuniah committed:
rating component && init form

keithCuniah committed:
init page

keithCuniah committed:
Merge branch 'releases/1.6.0' into feat/create-questionnaire-form-component

keithCuniah committed:
remove obsolete pages

keithCuniah committed:
add properties file

keithCuniah committed:
extract monoselection component && add textArea component

keithCuniah committed:
single label component

keithCuniah committed:
update form

keithCuniah committed:
Merge branch 'releases/1.6.0' into feat/create-questionnaire-form-component

keithCuniah committed:
conditional rendering in text2textContentEditable

keithCuniah committed:
textArea component css & logic

keithCuniah committed:
update form

keithCuniah committed:
add tooltip component

keithCuniah committed:
add tooltip icon in the 3 components

keithCuniah committed:
update param in form

keithCuniah committed:
textarea css

keithCuniah committed:
refacto props name

keithCuniah committed:
css && reset/validate buttons

keithCuniah committed:
manage error in different components

keithCuniah committed:
reset error state

keithCuniah committed:
update container

keithCuniah committed:
css icon info

keithCuniah committed:
move page exploration mode

keithCuniah committed:
init sidebar component

keithCuniah committed:
header for feedback task

Commit:
Merge branch 'feat/2615-instructions-task' into feat/create-questionnaire-form-component

Commit:
rename route to annotation-mode

Commit:
css form

Commit:
move questions form to content component

Commit:
min-width and scroll

leiyre commented on frontend/components/commons/forms/questions-form/fields/text-area/TextArea.component.vue:
```diff
@@ -0,0 +1,113 @@
+    </div>
+
+    <div class="container">
+      <Text2TextContentEditable
```
I would prefer to create a TextareaBase component for this case, because in this case always is editable and to avoid using business component

keithCuniah replied:
For now, I reuse things to match the goal to deliver something for the next week first iteration. But I agree that it will be nice to do a specific component for that in a refactoring after this iteration

leiyre commented on frontend/components/commons/header/header-feedback-task/HeaderFeedbackTask.component.vue:
```diff
@@ -0,0 +1,69 @@
+<template>
```
Is it not possible to use the standard header for all tasks? In case you need a specific one, better not to use the commons folder (for the components common to all tasks) and put it inside a folder (as a text classifier).

keithCuniah replied:
For later yes. But for the very first fast iterations, since we need to point to a new page with a new route it's better to use another header I think and then do the refactoring for homogenisation

leiyre commented on frontend/components/commons/sidebar-feedback-task/SidebarFeedbackTask.component.vue:
```diff
@@ -0,0 +1,21 @@
+<template>
+  <div class="sidebar"></div>
+</template>
```
Same, maybe we can create a feedback-task folder sibling to the other task (text-classifier) and put the components inside (to keep in commons only common components)

keithCuniah replied:
yes. This branch is only for the questionnaire but I put this div just to have a global view. When the sidebar development will start, it will be in a new branch.

Commit:
restore pre-commit file

keithCuniah committed:
feat : create layout instructions task (#2632)

# Description
create layout which will be use for instruction task dataset

Please include a summary of the changes and the related issue. Please
also include relevant motivation and context. List any dependencies that
are required for this change.

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

Co-authored-by: keithCuniah <keithcuniah@gmail.comh>

Commit:
Merge branch 'feat/front-instructions-task' into feat/create-questionnaire-form-component

Commit:
warning corrections in unit test

Commit:
Merge branch 'feat/2615-instructions-task' into feat/create-questionnaire-form-component

Commit:
import icon
